source/includes/usage-examples/code-snippets/distinct.go

```diff
@@ -39,16 +39,14 @@ func main() {
 
 	// Retrieves the distinct values of the "title" field in documents
 	// that match the filter
-	results, err := coll.Distinct(context.TODO(), "title", filter)
+	res := coll.Distinct(context.TODO(), "title", filter)
 
-	// Prints a message if any errors occur during the operation
-	if err != nil {
-		panic(err)
-	}
+	var arr []string
+	res.Decode(&arr)
```

Comment:
We still need to check the error here. Also, consider using the more common pattern of chaining the `Distinct` call.

For example:
```go
var arr []string
err := coll.Distinct(context.TODO(), "title", filter).Decode(&arr)
if err != nil {
	panic(err)
}
```

Reply:
thanks for clarifying the API, will change!